### PR TITLE
Update polkadot dependencies

### PR DESCRIPTION
**Pull request by vgeddes**
Updated Polkadot/Cumulus dependencies to polkadot v0.9.4. This also required us to update GSRPC to get metadata V13 changes.

Note that XCM verification & testing is pushed out to a follow-up PR given the scope-creep that has arisen so far.

E2E tests mostly pass. For the tests that fail, there is an unrelated & pre-existing issue in the commitment relayer which results in some commitments getting ignored and are not being passed on to Ethereum.

Changes:
* Parachain
  * Update deps, runtime code, and chain specs to work with polkadot v0.9.4
  * XCM: Remove obsolete `transfer` pallet. We no longer need a custom pallet to allow users to initiate XCM transfers.
  * Use Aura for block authoring
  * Block time increased to 12 secs to match reality around current relay chain performance.
* Relayer:
  * Update GSRPC and go-ethereum deps
  * Update receipt trie generation to work with go-ethereum 1.10.3
* GSRPC (branch https://github.com/Snowfork/go-substrate-rpc-client/tree/centrifuge-rebased):
  * Contains latest upstream changes
  * Various bug & polkadot v0.9.4 compatibility fixes for the following issues:
    *  https://github.com/paritytech/substrate/issues/9137
    * https://github.com/centrifuge/go-substrate-rpc-client/pull/171
    * https://github.com/centrifuge/go-substrate-rpc-client/issues/168
 * E2E Stack
   * Requires latest polkadot-launch.
   * Need to run 2 collator nodes (alice & bob) as a minimum.
   * Expose unsafe RPC methods (required for offchain DB queries in Polkadot v0.9.4)

### CODE REVIEW

File: parachain/runtime/local/src/lib.rs [hunk 1]
```diff
@@ -644,30 +644,35 @@ construct_runtime!(
...
-		VerifierLightclient: verifier_lightclient::{Pallet, Call, Storage, Event, Config} = 14,
-		Assets: assets::{Pallet, Call, Config<T>, Storage, Event<T>} = 15,
+		// Bridge Infrastructure
+		BasicInboundChannel: basic_channel_inbound::{Pallet, Call, Config, Storage, Event} = 10,
+		BasicOutboundChannel: basic_channel_outbound::{Pallet, Config<T>, Storage, Event} = 11,
+		IncentivizedInboundChannel: incentivized_channel_inbound::{Pallet, Call, Config, Storage, Event} = 12,
+		IncentivizedOutboundChannel: incentivized_channel_outbound::{Pallet, Config<T>, Storage, Event} = 13,
+		Dispatch: dispatch::{Pallet, Call, Storage, Event<T>, Origin} = 14,
+		VerifierLightclient: verifier_lightclient::{Pallet, Call, Storage, Event, Config} = 15,
+		Assets: assets::{Pallet, Call, Config<T>, Storage, Event<T>} = 16,
```
Comment (philipstanislaus) [inline parachain/runtime/local/src/lib.rs:662]: Shall we also move those indices to a higher starting index? If they shift in later upgrades, that would also be super annoying for anyone using the pallet, right?

File: parachain/runtime/local/src/lib.rs [hunk 2]
```diff
@@ -644,30 +644,35 @@ construct_runtime!(
...
-		Utility: pallet_utility::{Pallet, Call, Event, Storage} = 20,
+		// XCM
+		XcmpQueue: cumulus_pallet_xcmp_queue::{Pallet, Call, Storage, Event<T>} = 17,
+		DmpQueue: cumulus_pallet_dmp_queue::{Pallet, Call, Storage, Event<T>} = 18,
+		PolkadotXcm: pallet_xcm::{Pallet, Call, Event<T>, Origin} = 19,
+		CumulusXcm: cumulus_pallet_xcm::{Pallet, Origin} = 20,
 
 		// For dev only, will be removed in production
-		Sudo: pallet_sudo::{Pallet, Call, Config<T>, Storage, Event<T>} = 21,
+		Sudo: pallet_sudo::{Pallet, Call, Config<T>, Storage, Event<T>} = 20,
```
Comment (philipstanislaus) [inline parachain/runtime/local/src/lib.rs:671]: Sudo and CumulusXcm have the same index now?

File: parachain/runtime/snowbridge/src/lib.rs [hunk 3]
```diff
@@ -645,26 +622,26 @@ construct_runtime!(
...
-		Dispatch: dispatch::{Pallet, Call, Storage, Event<T>, Origin} = 13,
-		VerifierLightclient: verifier_lightclient::{Pallet, Call, Storage, Event, Config} = 14,
-		Assets: assets::{Pallet, Call, Config<T>, Storage, Event<T>} = 15,
+		BasicInboundChannel: basic_channel_inbound::{Pallet, Call, Config, Storage, Event} = 10,
+		BasicOutboundChannel: basic_channel_outbound::{Pallet, Config<T>, Storage, Event} = 11,
+		IncentivizedInboundChannel: incentivized_channel_inbound::{Pallet, Call, Config, Storage, Event} = 12,
+		IncentivizedOutboundChannel: incentivized_channel_outbound::{Pallet, Config<T>, Storage, Event} = 13,
+		Dispatch: dispatch::{Pallet, Call, Storage, Event<T>, Origin} = 14,
+		VerifierLightclient: verifier_lightclient::{Pallet, Call, Storage, Event, Config} = 15,
+		Assets: assets::{Pallet, Call, Config<T>, Storage, Event<T>} = 16,
```
Comment (philipstanislaus) [inline parachain/runtime/snowbridge/src/lib.rs:639]: As above

**Review by philipstanislaus — COMMENTED** [review]
Cool, few minor comments

File: parachain/runtime/local/src/lib.rs (hunk 2 above)
Comment (vgeddes) [reply parachain/runtime/local/src/lib.rs]: Nice catch! Fixes.

File: parachain/runtime/local/src/lib.rs (hunk 1 above)
Comment (vgeddes) [reply parachain/runtime/local/src/lib.rs:662]: Nah, not too worried about this for now. Most clients will dynamically lookup the pallet indices in the metadata.

Further down the line, when other parachains issue calls to us via XCM, then we may need to look at locking down these indices somehow.

File: parachain/runtime/local/src/lib.rs [hunk 4]
```diff
@@ -267,106 +281,163 @@ impl pallet_transaction_payment::Config for Runtime {
...
 	type Call = Call;
-	type XcmSender = LocalXcmHandler;
+	type XcmSender = XcmRouter;
+	// How to withdraw and deposit an asset.
 	type AssetTransactor = LocalAssetTransactor;
-	type OriginConverter = LocalOriginConverter;
+	type OriginConverter = XcmOriginToTransactDispatchOrigin;
 	type IsReserve = NativeAsset;
-	type IsTeleporter = ();
+	type IsTeleporter = NativeAsset; // <- should be enough to allow teleportation of ROC
```
Comment (philipstanislaus) [inline parachain/runtime/local/src/lib.rs:388]: Here we support teleportation though?

File: parachain/runtime/local/src/lib.rs [hunk 5]
```diff
@@ -267,106 +281,163 @@ impl pallet_transaction_payment::Config for Runtime {
...
 	type LocationInverter = LocationInverter<Ancestry>;
+	type Barrier = Barrier;
+	type Weigher = FixedWeightBounds<UnitWeightCost, Call>;
+	type Trader = UsingComponents<IdentityFee<Balance>, RococoLocation, AccountId, Balances, ()>;
+	type ResponseHandler = (); // Don't handle responses for now.
```
Comment (philipstanislaus) [inline parachain/runtime/local/src/lib.rs:393]: What's the reason for this?

**Review by philipstanislaus — APPROVED** [review]
Nice, just few questions 💪

File: parachain/runtime/local/src/lib.rs [hunk 6]
```diff
@@ -267,106 +281,163 @@ impl pallet_transaction_payment::Config for Runtime {
...
 	Balances,
 	// Use this currency when it is a fungible asset matching the given location or name:
 	IsConcrete<RococoLocation>,
 	// Do a simple punn to convert an AccountId32 MultiLocation into a native chain account ID:
-	LocationConverter,
+	LocationToAccountId,
 	// Our chain's account ID type (we can't get away without mentioning it explicitly):
 	AccountId,
+	// We don't track any teleports.
+	()
```
[the remark this answers is not in the document]
Comment (vgeddes) [reply parachain/runtime/local/src/lib.rs:334]: By design, we don't support teleportation.  Only reserve-backed transfers, which doesn't require trust in the other chain.

File: parachain/runtime/local/src/lib.rs (hunk 5 above)
Comment (vgeddes) [reply parachain/runtime/local/src/lib.rs:393]: No idea. 😅 

XCM documentation is sparse and a lot of the new XCM code I copy-pasted from the Cumulus parachains at https://github.com/paritytech/cumulus/tree/master/polkadot-parachains (statemint, rococo examples, etc)

File: parachain/runtime/local/src/lib.rs (hunk 4 above)
Comment (vgeddes) [reply parachain/runtime/local/src/lib.rs:388]: Good catch, will fix!

**Pull request merged**